core/remotes/docker/fetcher.go

```diff
@@ -262,7 +264,27 @@ func (r dockerFetcher) FetchByDigest(ctx context.Context, dgst digest.Digest, op
 	return seeker, desc, nil
 }
 
+const (
+	_        = iota
+	kB int64 = 1 << (10 * iota)
+	mB
+	gB
+	tB
+	pB
+	eB
+
+	// When under this size, a layer will not be fetched using parallel requets
+	minimumSizeForParallelDL = 1 * mB
+)
+
 func (r dockerFetcher) open(ctx context.Context, req *request, mediatype string, offset int64) (_ io.ReadCloser, retErr error) {
+	parallelism, chunkSize := r.config.Parallelism(), int64(r.config.ConcurrentFetchChunksSizeMB)*mB
+	if r.config.Limiter != nil {
+		if err := r.config.Limiter.Acquire(ctx, parallelism); err != nil {
```

Comment:
So we won't start doing any work until we can fire on all cylinders? Would it make sense to use `TryAcquire` and accept a lower parallelism if we can avoid waiting?

Reply:
That would work but we would have a chance of downloading the biggest layer of the whole set without going full throttle. Which could randomly lower speed by a lot. Trying to see if I can have a setup were we can have both.